flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -230,13 +233,34 @@ public GetShardListResult getShardList(Map<String, String> streamNamesWithLastSe
 	 */
 	@Override
 	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, @Nullable String startingSeqNum) throws InterruptedException {
+		GetShardIteratorRequest getShardIteratorRequest = new GetShardIteratorRequest()
+			.withStreamName(shard.getStreamName())
+			.withShardId(shard.getShard().getShardId())
+			.withShardIteratorType(shardIteratorType)
+			.withStartingSequenceNumber(startingSeqNum);
+		return getShardIterator(getShardIteratorRequest);
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, @Nonnull final Date startingTimestamp) throws InterruptedException {
+		GetShardIteratorRequest getShardIteratorRequest = new GetShardIteratorRequest()
+			.withStreamName(shard.getStreamName())
+			.withShardId(shard.getShard().getShardId())
+			.withShardIteratorType(shardIteratorType)
+			.withTimestamp(startingTimestamp);
```

Comment:
How does the Kinesis API behave, when you set both a non-matching shard iterator type, and a timestamp?
In other words, what happens when the shard iterator type is perhaps `TRIM_HORIZON`, but a timestamp is also provided?

If the API call will fail with Kinesis doesn't allow such combinations, we probably should make this method implementation fail-proof of such situations.